packages/vector_graphics/lib/src/listener.dart

```diff
@@ -52,8 +52,19 @@ PictureInfo decodeVectorGraphics(
     locale: locale,
     textDirection: textDirection,
   );
-  _codec.decode(data, listener);
-  return listener.toPicture();
+  DecodeResponse response = _codec.decode(data, listener);
+  if (response.complete) {
+    final PictureInfo picture = listener.toPicture();
```

Comment:
unrelated: should we just make `toPicture` destructive and cause disposal to happen?

Reply:
Yeah, simpler - done.